File: domain/src/test/java/com/arturogutierrez/openticator/domain/otp/OneTimePasswordGeneratorTest.java
package com.arturogutierrez.openticator.domain.otp;

import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;

public class OneTimePasswordGeneratorTest {

  @Test
  public void testGenerateCodeForFirstState() {
    String secret = "ABCDEFGHIJK23456";
    OneTimePasswordGenerator oneTimePasswordGenerator = new OneTimePasswordGenerator(secret, 6);

    String code = oneTimePasswordGenerator.generate(0);

    assertThat(code, is("867671"));
  }

  @Test
  public void testGenerateCodeForNinthState() {
    String secret = "ABCDEFGHIJK23456";
    OneTimePasswordGenerator oneTimePasswordGenerator = new OneTimePasswordGenerator(secret, 6);

    String code = oneTimePasswordGenerator.generate(9);

    assertThat(code, is("877436"));
  }
}

File: domain/src/test/kotlin/com/arturogutierrez/openticator/domain/otp/OneTimePasswordGeneratorTest.kt
package com.arturogutierrez.openticator.domain.otp

import com.natpryce.hamkrest.assertion.assertThat
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class OneTimePasswordGeneratorTest {

    @Test
    fun testGenerateCodeForFirstState() {
        val secret = "ABCDEFGHIJK23456"
        val oneTimePasswordGenerator = OneTimePasswordGenerator(secret, 6)

        val code = oneTimePasswordGenerator.generate(0)

        assertThat(code, equalTo("867671"))
    }

    @Test
    fun testGenerateCodeForNinthState() {
        val secret = "ABCDEFGHIJK23456"
        val oneTimePasswordGenerator = OneTimePasswordGenerator(secret, 6)

        val code = oneTimePasswordGenerator.generate(9)

        assertThat(code, equalTo("877436"))
    }
}
